sympy/solvers/solveset.py

```diff
@@ -855,6 +863,9 @@ def solveset(f, symbol=None):
 
     f = sympify(f)
 
+    if f is S.false:
+        return EmptySet()
```

Comment:
> But Is not solveset supposed to return always a set ?

Yes, perhaps this is a good catch, but not related to this PR. This could be incorporated in a different commit.